### PR TITLE
Units change and maintaing of aspect ratio

### PR DESCRIPTION
**Pull request by lisham2000**
Added units to SVG export 
Added that it maintains aspect ratio and grays out other dimension not being specified
Solution #1051

### CODE REVIEW

File: nion/swift/ExportDialog.py [hunk 1]
```diff
@@ -239,45 +239,133 @@ def export_clicked(display_items: typing.Sequence[DisplayItem.DisplayItem], view
...
         super().__init__()
-
         self.__document_controller = document_controller
-
         u = Declarative.DeclarativeUI()
-
         if display_item.display_data_shape and len(display_item.display_data_shape) == 2:
-            display_size = Geometry.IntSize(height=4, width=4)
+            display_size = Geometry.IntSize(height=display_item.display_data_shape[1],
+                                            width=display_item.display_data_shape[0])
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:361]: The dimension of `display_size` are swapped. We use the numpy convention: height is `[0]` and width is `[1]`. Try exporting a 10x100 and 100x10 data item.

File: nion/swift/ExportDialog.py [hunk 2]
```diff
@@ -239,45 +239,133 @@ def export_clicked(display_items: typing.Sequence[DisplayItem.DisplayItem], view
...
         else:
-            display_size = Geometry.IntSize(height=3, width=4)
-
+            display_size = Geometry.IntSize(height=4, width=4)
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:363]: For line plots (1D data), use a default size of `width=5, height=3`

File: nion/swift/ExportDialog.py [hunk 3]
```diff
@@ -302,6 +390,7 @@ def cancel_clicked() -> bool:
         dialog.show()
 
 
+
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:393]: Remove this whitespace/blank line change.

File: nion/swift/ExportDialog.py [hunk 4]
```diff
@@ -239,45 +239,133 @@ def export_clicked(display_items: typing.Sequence[DisplayItem.DisplayItem], view
...
+        )
+        units_row = u.create_row(
+            u.create_label(text=_("Units:"), width=80),
+            self.units_combo_box,
+            spacing=12
+        )
+        main_page = u.create_column(width_row, height_row, units_row, spacing=12, margin=12)
         self.ui_view = main_page
 
+    def value_updated(self, widget: Declarative.UIWidget, text: str) -> None:
```
Comment (cmeyer) [inline nion/swift/ExportDialog.py:291]: Hmmm. I seem to have missed this in a bunch of other code, but let's start now!

Widgets passed to declarative callback functions are the actual widgets defined in `UserInterface.py`. So this function should be:

`def value_updated(self, widget: UserInterface.LineEditWidget, text: str) -> None:`

This goes for all similar cases in this code.

**Review by cmeyer — CHANGES_REQUESTED** [review]
Thanks for the PR! I like how it works. However, I've requested several changes to make the code easier to maintain.

Also note: there is no reason to "force push" or rebase for this PR - just make new commits and regular "push" them to this PR unless you have another reason to rebase from master. That will make it somewhat easier for me to review requested changes.

**Comment by lisham2000** [comment]
> uested several changes to make the code easier to maintain.
> 
> Also note: there is no reason to "force push" or rebase for this PR - just make new commits and regular "push" them to t

Hi Chris , the reason for the force push was having to squash since  my code seemed to pass relevant checks on my local version of Mypy when i went to push it took me multiple commits for it to pass the online one.

**Comment by cmeyer** [comment]
This has been succeeded by #1100, right? Can we close this PR?

**Comment by lisham2000** [comment]
Closed due to revised version at #1106

**Pull request closed without merging**